collect/collect.go

```diff
@@ -196,68 +195,46 @@ func (i *InMemCollector) AddSpanFromPeer(sp *types.Span) {
 // block is the only place we are allowed to modify any running data
 // structures.
 func (i *InMemCollector) collect() {
-	// make sure we send all pending traces when we finish
-	defer func() {
-		for sig := range i.toSend {
-			i.send(sig.trace)
-		}
-	}()
-
-	peerChanSize := cap(i.fromPeer)
+	ticker := time.NewTicker(i.Config.GetSendTickerValue())
+	defer ticker.Stop()
+
+	incoming := mergeIncomingSpans(spanInput{
+		ch:          i.incoming,
+		name:        "from_incoming",
+		concurrency: 1,
+	}, spanInput{
+		ch:          i.fromPeer,
+		name:        "from_peer",
+		concurrency: 2,
+	})
```

Comment:
Another possible resolution - instead of priority draining the peer queue, what about pausing the incoming queue? Give the system a way to say at 80% tell the incoming queue to pause work until we get below.